Test/Integration/Model/ScriptGeneratorTest.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+declare(strict_types=1);
+
+namespace CustomGento\Cookiebot\Test\Integration\Model;
+
+use CustomGento\Cookiebot\Model\Config;
+use CustomGento\Cookiebot\Model\ScriptGenerator;
+use Magento\Framework\App\ObjectManager;
+use Magento\TestFramework\ObjectManager as TestFrameworkObjectManager;
+use PHPUnit\Framework\TestCase;
+
+class ScriptGeneratorTest extends TestCase
+{
+    /**
+     * @var Config
+     */
+    private $config;
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+        $this->config = TestFrameworkObjectManager::getInstance()->create(Config::class);
+    }
+
+    /**
+     * @magentoConfigFixture current_store web/cookiebot/use_eu_cdn 0
+     */
+    public function testWhenUseEuCdnIsDisabled(): void
+    {
+        $generatedScript = ObjectManager::getInstance()->create(ScriptGenerator::class)->generate();
+        self::assertEquals('<script id="Cookiebot" data-cfasync="false" src="https://consent.cookiebot.com/uc.js" data-cbid=""  type="text/javascript" async></script>',
```

Comment:
I don't think this can pass as it includes a duplicate whitespace after the data-cbid attribute, which should not be in the generated script?
Anyway, since only the domain changes, I think it would be enough to check if the right domain is included in the script?

Reply:
No, there should be exactly two space to be passed, you mean I should first get the url by regex and then check the URL? Don't you think in this way the performance decreased and we should first get the URL?